backend/gateway/log/log.go

```diff
@@ -16,3 +18,23 @@ func ProtoField(key string, m proto.Message) zap.Field {
 	}
 	return zap.Any(key, json.RawMessage(b))
 }
+
+// GrpcStatusField takes in an error and returns a zap field that logs as the embedded JSON
+// representation of a gRPC Status message.
+func GrpcStatusField(key string, err error) zap.Field {
+	var m *spb.Status
+	// the second return val is a bool reporting whether it was able to
+	// convert the input error to a Status. If false, FromError returns a new Status with
+	// codes.Unknown and the original input error message.
+	// https: //github.com/grpc/grpc-go/blob/master/status/status.go#L81
+	s, _ := status.FromError(err)
```

Comment:
i'd still return the original error in the event it's not a gRPC error. then we can use this function everywhere without worrying about whether or not it's a status and that changing the error up to appear as if it was a status. and for that matter we can rename this function to NamedError and create an Error helper just like zap has to be used everywhere.
```suggestion
	s, ok := status.FromError(err)
	if !ok {
	   return zap.NamedError(key, err)
	}
```

Reply:
got it, yeah that would be better. will update

Reply:
updated the local testing examples in the PR description